datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -1725,6 +1774,27 @@ mod tests {
             DataType::LargeBinary
         );
 
+        // Timestamps
+        let utc: Option<Arc<str>> = Some("UTC".into());
+        test_coercion_binary_rule!(
+            DataType::Timestamp(TimeUnit::Second, utc.clone()),
+            DataType::Timestamp(TimeUnit::Second, utc.clone()),
+            Operator::Eq,
+            DataType::Timestamp(TimeUnit::Second, utc.clone())
+        );
+        test_coercion_binary_rule!(
+            DataType::Timestamp(TimeUnit::Second, utc.clone()),
+            DataType::Timestamp(TimeUnit::Second, Some("Europe/Brussels".into())),
+            Operator::Eq,
+            DataType::Timestamp(TimeUnit::Second, utc.clone())
+        );
```

Comment:
Maybe we could also add a test showing that the lhs is still picked even with UTC

Like this I think would pass.


```rust
        test_coercion_binary_rule!(
            DataType::Timestamp(TimeUnit::Second, Some("Europe/Brussels".into())),
            DataType::Timestamp(TimeUnit::Second, utc.clone()),
            Operator::Eq,
            DataType::Timestamp(TimeUnit::Second, Some("Europe/Brussels".into())),
        );
```

Though I am a little confused if this is the intention (or if this case was supposed to coerce to UTC)

Reply:
This would pass as the code is currently written. I was originally leaning towards coercing to UTC if it existed, otherwise choosing the left. But I was worried that may be a little confusing and settled on always coercing to the left. I'm happy to change it if we'd prefer the other way though.

Reply:
I think we should add the test so the behavior is clear

I also think the current behavior as implemented in this PR (coerce to the left) is a good idea